lib/travis/build/git/clone.rb

```diff
@@ -22,12 +22,22 @@ def apply
 
           def clone_or_fetch
             sh.if "! -d #{dir}/.git" do
-              sh.cmd "git clone #{clone_args} #{data.source_url} #{dir}", assert: true, retry: true
-              if github?
-                sh.if "$? -ne 0" do
-                  sh.echo "Failed to clone from GitHub.", ansi: :red
-                  sh.echo "Checking GitHub status (https://status.github.com/api/last-message.json):"
-                  sh.raw "curl -sL https://status.github.com/api/last-message.json | jq -r .[]"
+              if sparse_checkout
+                sh.cmd "git init #{dir}", assert: true, retry: true
+                sh.cmd "git -C #{dir} config core.sparseCheckout true", assert: true, retry: true
+                sh.cmd "echo #{sparse_checkout} >> #{dir}/.git/info/sparseCheckout", assert: true, retry: true
+                sh.cmd "git -C #{dir} remote add origin #{data.source_url}", assert: true, retry: true
+                sh.cmd "git -C #{dir} pull origin #{branch} #{pull_args}", assert: true, retry: true
+                sh.cmd "cat #{sparse_checkout} >> #{dir}/.git/info/sparseCheckout", assert: true, retry: true
```

Comment:
@PeterDaveHello Is this `cat` correct? I don't think there is a file with this name exists at this point in execution.

Reply:
Indeed, it fails when the execution actually gets to this spot: https://staging.travis-ci.org/BanzaiMan/travis_staging_test/builds/617995